src/Misc/Part.cpp

```diff
@@ -165,8 +165,19 @@ static const Ports partPorts = {
 
             //d.broadcast("/damage", "s", part_loc);
         }},
-
-
+        {"savexml:", rProp(internal) rDoc("Save Part to the file it has been loaded from"), 0,
+        [](const char *, RtData &d)
+        {
+            Part *p = (Part*)d.obj;
+            if (!p->hasFilename) {  // if part was never loaded or saved
+                time_t rawtime;     // make a new name from date and time
+                struct tm * timeinfo;
+                time (&rawtime);
+                timeinfo = localtime (&rawtime);
```

Comment:
You could just write `const struct tm* timeinfo = localtime (&rawtime);` to save a line of code.

Reply:
done